bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/StreamServerHandler.java

```diff
@@ -177,19 +177,13 @@ public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object ms
             }
             if (msg instanceof HttpContent) {
                 HttpContent content = (HttpContent) msg;
-                int index = 0;
                 if (recievedBytes == 0) {
-                    incomingJpeg = new byte[content.content().capacity()];
+                    incomingJpeg = content.content().array();
                 } else {
                     byte[] temp = incomingJpeg;
                     incomingJpeg = new byte[recievedBytes + content.content().capacity()];
-
-                    for (; index < temp.length; index++) {
-                        incomingJpeg[index] = temp[index];
-                    }
-                }
-                for (int i = 0; i < content.content().capacity(); i++) {
-                    incomingJpeg[index++] = content.content().getByte(i);
+                    System.arraycopy(temp, 0, incomingJpeg, 0, temp.length);
+                    System.arraycopy(content.content(), 0, incomingJpeg, recievedBytes, content.content().capacity());
```

Comment:
`content.content()` isn't an array type, so I don't think `System.arraycopy` would work as written.
```suggestion
                    System.arraycopy(content.content().array(), 0, incomingJpeg, recievedBytes, content.content().capacity());
```